idea/idea-fir/src/org/jetbrains/kotlin/idea/fir/intentions/HLAddAccessorIntentions.kt

```diff
@@ -0,0 +1,68 @@
+import org.jetbrains.kotlin.idea.api.applicator.HLApplicatorInput
+import org.jetbrains.kotlin.idea.api.applicator.applicator
+import org.jetbrains.kotlin.idea.fir.api.AbstractHLIntention
+import org.jetbrains.kotlin.idea.fir.api.applicator.HLApplicabilityRange
+import org.jetbrains.kotlin.idea.fir.api.applicator.HLApplicatorInputProvider
+import org.jetbrains.kotlin.idea.fir.api.applicator.inputProvider
+import org.jetbrains.kotlin.idea.fir.applicators.ApplicabilityRanges
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtPropertySymbol
+import org.jetbrains.kotlin.idea.intentions.AbstractAddAccessorsIntention
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.load.java.JvmAbi
+import org.jetbrains.kotlin.name.ClassId
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.psiUtil.containingClass
+import org.jetbrains.kotlin.psi.psiUtil.containingClassOrObject
+import org.jetbrains.kotlin.psi.psiUtil.hasExpectModifier
+
+abstract class HLAddAccessorIntention(private val addGetter: Boolean, private val addSetter: Boolean) :
+    AbstractHLIntention<KtProperty, HLApplicatorInput.Empty>(KtProperty::class) {
+    // TODO: Copy and test range from AddAccessorIntentions.applicabilityRange
+    override val applicabilityRange: HLApplicabilityRange<KtProperty> = ApplicabilityRanges.SELF
+
+    override val applicator: HLApplicator<KtProperty, HLApplicatorInput.Empty> = applicator {
+        applyTo { ktProperty, _, _, editor ->
+            AbstractAddAccessorsIntention.applyTo(ktProperty, editor, addGetter, addSetter)
+        }
+        familyAndActionName(AbstractAddAccessorsIntention.createFamilyName(addGetter, addSetter))
+    }
+
+    override val inputProvider: HLApplicatorInputProvider<KtProperty, HLApplicatorInput.Empty> = inputProvider { ktProperty ->
+        if (ktProperty.isLocal || ktProperty.hasDelegate() ||
+            ktProperty.containingClass()?.isInterface() == true ||
+            ktProperty.containingClassOrObject?.hasExpectModifier() == true ||
+            ktProperty.hasModifier(KtTokens.ABSTRACT_KEYWORD) ||
+            ktProperty.hasModifier(KtTokens.LATEINIT_KEYWORD) ||
+            ktProperty.hasModifier(KtTokens.CONST_KEYWORD)
+        ) {
+            return@inputProvider null
+        }
+
+        val symbol = ktProperty.getVariableSymbol() as? KtPropertySymbol ?: return@inputProvider null
+        if (symbol.containsAnnotation(JVM_FIELD_CLASS_ID)) return@inputProvider null
+
+        val hasInitializer = ktProperty.hasInitializer()
+        if (ktProperty.typeReference == null && !hasInitializer) return@inputProvider null
+        if (addSetter && (!ktProperty.isVar || ktProperty.setter != null)) return@inputProvider null
+        if (addGetter && ktProperty.getter != null) return@inputProvider null
+
+        HLApplicatorInput.Empty
+    }
+
+    companion object {
+        private val JVM_FIELD_CLASS_ID = ClassId.topLevel(JvmAbi.JVM_FIELD_ANNOTATION_FQ_NAME)
```

Comment:
Please, consider adding it to `org.jetbrains.kotlin.name.StandardClassIds`

Reply:
Currently there are only non-platform specific IDs in there. I'm leaning towards keeping it that way if it's not a significant concern. Unfortunately, the JVM class IDs are all over the place (just looking at usage of `ClassId.topLevel`) and there is not a single spot for JVM class IDs.